core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -160,7 +163,8 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is not started")
     } else {
       val scheme = if (securityManager.fileServerSSLOptions.enabled) "https" else "http"
-      s"$scheme://${Utils.localIpAddress}:$port"
+      val hosturi = InetAddresses.toUriString(InetAddress.getByName(localHostname))
+      s"$scheme://${InetAddresses.toUriString(hosturi)}:$port"
```

Comment:
Now `toUriString` is called twice -- the string should just refer to `${hosturi}`.
While we're doing another last pass, two more tiny nits: `hostUri` instead of `hosturi` at least, and, there is a spurious blank line added in the other file.